pkg/pushsync/pushsync.go

```diff
@@ -195,25 +270,41 @@ func (ps *PushSync) chunksWorker() {
 	}
 }
 
-// sendChunkMsg sends a chunk to a given peer.
-func (ps *PushSync) sendChunkMsg(ctx context.Context, peer swarm.Address, ch swarm.Chunk) error {
-	startTimer := time.Now()
+// sendChunkAndReceiveReceipt sends chunk to a given peer
+// by opening a stream. It then waits for a receipt from that peer.
+// Once the receipt is received within a given time frame it marks that this chunk
+// as synced in the localstore.
+func (ps *PushSync) SendChunkAndReceiveReceipt(ctx context.Context, peer swarm.Address, ch swarm.Chunk) error {
 	streamer, err := ps.streamer.NewStream(ctx, peer, nil, protocolName, protocolVersion, streamName)
 	if err != nil {
 		return fmt.Errorf("new stream: %w", err)
 	}
 	defer streamer.Close()
 
-	w, _ := protobuf.NewWriterAndReader(streamer)
+	w, r := protobuf.NewWriterAndReader(streamer)
+	if err := ps.sendChunkDelivery(w, ch); err != nil {
+		return fmt.Errorf("could not send chunk delivery : %w", err)
```

Comment:
The error annotation is better without `could not` in it as errors are chained.

There is a blank space before the colon that should be removed.